src/Iact1dUnbinnedLkl.cc

```diff
@@ -289,66 +289,77 @@ Int_t Iact1dUnbinnedLkl::InterpretInputString(TString inputString)
     }
   
   // open and read input files with data and IRFs
-  TFile* ifile = new TFile(path+(path==""?"":"/")+inputfileName,"READ");
-  IactEventListIrf* dataSet = (IactEventListIrf*) ifile->Get("IactEventListIrf");
-  if(!dataSet)
+  IactEventListIrf* dataSet = new IactEventListIrf();
+  if(inputfileName.EndsWith(".root"))
     {
-      cout << "Iact1dUnbinnedLkl::InterpretInputString Warning: no IactEventListIrf object in file " << inputfileName << ", trying the CTAIRF format" << endl;
-      delete ifile; ifile=NULL;
-      if(ReadCTAIRF(path+"/"+inputfileName))
-	return 1;
+      TFile* ifile = new TFile(path+(path==""?"":"/")+inputfileName,"READ");
+      dataSet = (IactEventListIrf*) ifile->Get("IactEventListIrf");
+      if(!dataSet) // check if the ROOT file was correctly interpreted
+        {
+          cout << "Iact1dUnbinnedLkl::InterpretInputString Warning: no IactEventListIrf object in file " << inputfileName << ", trying the CTAIRF format" << endl;
+          if(ReadCTAIRF(path+"/"+inputfileName)) 
+            return 1;
+        }
+      ifile->Close();
+      delete ifile;
     }
+  else if(inputfileName.EndsWith(".fits"))
+    {
```

Comment:
According to the optimal [i.e. mine] style, brackets and stuff should not be included for single-line blocks...